airflow/contrib/operators/emr_add_steps_operator.py

```diff
@@ -40,18 +43,26 @@ class EmrAddStepsOperator(BaseOperator):
     @apply_defaults
     def __init__(
             self,
-            job_flow_id,
+            job_flow_id=None,
+            job_flow_name=None,
             aws_conn_id='s3_default',
             steps=None,
             *args, **kwargs):
         super().__init__(*args, **kwargs)
         steps = steps or []
-        self.job_flow_id = job_flow_id
         self.aws_conn_id = aws_conn_id
+        self.emr_hook = EmrHook(aws_conn_id=self.aws_conn_id)
```

Comment:
Please don't do this -- don't make connections/requests inside `__init__`, it should only be done in `execute()`